src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -1165,21 +1166,37 @@ BOOST_AUTO_TEST_CASE(SimpleSigmoidCrossEntropyLayerTest)
   double error2 = module.Forward(std::move(input2), std::move(target2));
   BOOST_REQUIRE_SMALL(error2 / input2.n_elem - 1.5027283, 1e-6);
 
+  input3 = arma::mat("0 -1 -1 0 -1 0 0 -1");
+  target3 = arma::mat("0 -1 -1 0 -1 0 0 -1");
+  double error3 = module.Forward(std::move(input3), std::move(target3));
+  BOOST_REQUIRE_SMALL(error3 / input3.n_elem - 0.00320443, 1e-6);
+
   // Test the Backward function.
   module.Backward(std::move(input1), std::move(target1), std::move(output));
   for (size_t i = 0; i < output.n_elem; i++)
-    BOOST_REQUIRE_SMALL(output(i) - 0.62245929, 1e-5);
+    BOOST_REQUIRE_SMALL(output(i) - 0.37754066, 1e-5);
   BOOST_REQUIRE_EQUAL(output.n_rows, input1.n_rows);
   BOOST_REQUIRE_EQUAL(output.n_cols, input1.n_cols);
 
   expectedOutput = arma::mat(
-      "0.7310586 0.88079709 -0.04742587 0.98201376 -0.00669285");
+      "0.2689414 0.1192029 -0.04742587 0.0179862 -0.00669285");
   module.Backward(std::move(input2), std::move(target2), std::move(output));
   for (size_t i = 0; i < output.n_elem; i++)
     BOOST_REQUIRE_SMALL(output(i) - expectedOutput(i), 1e-5);
-
   BOOST_REQUIRE_EQUAL(output.n_rows, input2.n_rows);
   BOOST_REQUIRE_EQUAL(output.n_cols, input2.n_cols);
+
+  module.Backward(std::move(input3), std::move(target3), std::move(output));
+  for (size_t i = 0; i < 8; ++i)
+  {
+    double el = output.at(0, i);
+    if (std::abs(input3.at(i) - 0.0) < 1e-5)
+      BOOST_REQUIRE_SMALL(el - 0.5, 2e-6);
+    else
+      BOOST_REQUIRE_SMALL(el - 1.2689414, 2e-6);
+  }
```

Comment:
I also don't follow where these numbers came from, could you add a comment describing why this is the correct output please?

Reply:
I checked them using `tensorflow.nn.sigmoid_cross_entropy_with_logits()`. The comment is added [here](https://github.com/mlpack/mlpack/pull/1200/files#diff-d686e5871ec51f01e597deaae104def3R1161)